docs/design/proposals/labels.md

```diff
@@ -0,0 +1,109 @@
+# Proposal: Standardize KRO Labels
+
+## Problem statement
+
+The current labels in KRO are used to track ownership and metadata of resources managed by the orchestrator. While these labels provide essential information, they lack a clear and standardized way to represent the relationships between different KRO components, such as ResourceGraphDefinitions (RGDs), instances, and the resources they generate. This ambiguity can make it difficult to query for related objects and understand the provenance of resources within a cluster.
+
+Specifically, we need a clearer way to identify:
+1.  Which RGD is responsible for reconciling a given instance.
+2.  Which RGD a created resource originates from.
+3.  Which instance a created resource belongs to.
+
+This proposal aims to introduce new, more descriptive labels to clarify these relationships, aligning with Kubernetes best practices for resource management and observability.
+
+## Proposal
+
+We propose introducing a set of standardized labels to better represent the relationships between RGDs, instances, and the resources they manage. This will improve the ability to discover, manage, and debug resources created and managed by KRO.
+
+#### Overview
+
+The proposal is to add the following labels:
+-   A label on an instance to indicate which RGD is reconciling it.
+-   A label on resources created by an instance reconciliation to indicate which RGD they come from.
+-   Labels on resources created by an instance reconciliation to indicate which instance created them.
+
+This aligns with the common Kubernetes practice of using labels for discovery and organization of related resources, as seen in tools like Helm and Kustomize.
+
+#### Design details
```

Comment:
should we have  a section addressing deeply nested RGDs (3+ levels)? thinking what happens when RGD1 creates an RGD2 instance, which creates an RGD3 instance, which creates an RGD4 instance ? With the two label scheme, we lose the intermediate relationships e.g an RGD4 instance would only show it was defined by RGD3 and controlled by RGD4, with no trace of RGD1 or RGD2. this means queries like "show me all resources that originated from RGD1, including those created indirectly through nested RGDs" won't be possible - which is fine, but worth noting.

Reply:
We could get around this by having a root RGD, although this starts to get reaaaally complicated by then :D